src/components/dialog/useFocusTrap.js

```diff
@@ -0,0 +1,82 @@
+import React, { useEffect, useRef, useCallback } from 'react';
+
+import { focusOnFirstDescendent, focusOnLastDescendent } from './utils';
+
+const FocusRing = ({ topFocusBumperRef, bottomFocusBumperRef, children }) => {
+  return (
+    <>
+      <div ref={topFocusBumperRef} tabIndex="0" />
+      {children}
```

Comment:
Hmm, I don't think that would solve breaking the :first-child and :last-child selectors though, since you're still adding a wrapping div?

Reply:
If you have this for example:

```jsx
<Popover>
  <div/>
  <div/>
  <div/>
</Popover>

```

With current implementation those pseudo selectors would change, because you're adding a different first and last child.

Reply:
I see, I'll add a wrapping div in this case!